src/components/forms/profile/update-user-address-form.tsx

```diff
@@ -0,0 +1,136 @@
+import React from "react";
+import { v4 as uuidv4 } from "uuid";
+
+import { InputText } from "../../input/input-text";
+// import { WrongInputError } from "../input/wrong-input-error";
+import { Form } from "../form";
+import { Address } from "@src/@types/profile";
+import { Button, ButtonVariant } from "@src/components/buttons/buttons";
+import { FakeButton } from "@src/components/buttons/fake-button";
+import { NeutralLink } from "@src/components/neutral-link/neutral-link";
+
+const UpdateUserAddressForm: React.FC<{
+  userAddress?: Address;
+}> = ({ userAddress }) => {
+  const [formID, setFormID] = React.useState<string>(uuidv4());
+
+  const [updatedAddress, setUpdatedAddress] = React.useState<Address>({
+    id: "",
+    sub: "",
+    street_address: "string",
+    locality: "string",
+    region: "string",
+    postal_code: "string",
+    country: "string",
+    kind: "string",
+    created_at: "string",
+    updated_at: "string",
+    street_address_2: "string",
+    primary: false,
+  });
+
+  React.useEffect(() => {
+    if (userAddress) {
+      setUpdatedAddress(userAddress);
+    }
+  }, [userAddress]);
+
+  return (
+    <>
+      <Form
+        formID={formID}
+        onSubmit={async () => {
+          setFormID(uuidv4());
+          return;
+        }}
+      >
+        <InputText
+          type="text"
+          name="street-address"
+          placeholder="Enter your street address"
+          value={updatedAddress.street_address}
+          onChange={(value) => {
+            setUpdatedAddress({
+              ...updatedAddress,
+              street_address: value,
+            });
+          }}
+          label="Street address *"
```

Comment:
is `*` used for "required"? 🤣

Reply:
If yes, could we make something like that?
```typescript
type InputTextProps = {
  /** the other props */
  required?: boolean
};

const InputText: React.FC<InputTextProps> = (props) => {
  const { label, className, children } = props;

  const inputRef = React.useRef(null);
  const { labelProps, inputProps } = useTextField(props, inputRef);

  return (
    <Label {...labelProps}>
      {label + required ? " *" : ""}
      {children ? (
        <ChildrenWrapper>
          {children}
          <InputElement {...inputProps} ref={inputRef} className={className} />
        </ChildrenWrapper>
      ) : (
        <InputElement {...inputProps} ref={inputRef} className={className} />
      )}
    </Label>
  );
};
```

Reply:
Well, we could, but like I said, I made a task to discuss validation and stuff. The `*` part was just following the figma mock.

Reply:
👍